### PR TITLE
Refactor to nva-commons

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: src/main/java/no/unit/nva/amazon/s3/CreateUploadHandler.java [hunk 1]
```diff
@@ -1,162 +1,115 @@
...
+
+    @Override
+    protected CreateUploadResponseBody processInput(CreateUploadRequestBody input, RequestInfo requestInfo,
+                                                    Context context) throws ApiGatewayException {
+        validate(input);
+
+        String keyName = UUID.randomUUID().toString();
+        InitiateMultipartUploadRequest initRequest =
+                new InitiateMultipartUploadRequest(bucketName, keyName, toObjectMetadata(input));
+        InitiateMultipartUploadResult initResponse = s3Client.initiateMultipartUpload(initRequest);
```
Comment (janbj) [inline src/main/java/no/unit/nva/amazon/s3/CreateUploadHandler.java:92]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code manually manages uploads or downloads. The SDK can automatically handle these operations. Consider using the SDK to automatically handle this instead. SDK class: TransferManager. See [Using TransferManager for Amazon S3 Operations](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/examples-s3-transfermanager.html).

File: build.gradle [hunk 2]
```diff
@@ -11,19 +11,28 @@ version '1.0-SNAPSHOT'
...
     implementation group: 'com.google.code.gson', name: 'gson', version: '2.8.6'
     implementation group: 'org.slf4j', name: 'slf4j-api', version: '1.7.29'
     implementation group: 'org.jsoup', name: 'jsoup', version: '1.12.1'
     implementation group: 'com.amazonaws', name: 'aws-lambda-java-core', version: '1.2.0'
     implementation group: 'com.amazonaws', name: 'aws-java-sdk-s3', version: '1.11.715'
+    implementation group: 'org.glassfish.jersey.core', name: 'jersey-common', version: '2.22.2'
+    implementation group: 'org.zalando', name: 'problem', version: '0.23.0'
+    implementation group: 'org.zalando', name: 'jackson-datatype-problem', version: '0.23.0'
 
+    testImplementation group: 'org.glassfish.jersey.core', name: 'jersey-common', version: '2.22.2'
```
Comment (axthosarouris) [inline build.gradle:34]: why do you need to repeat the dependency?

File: src/main/java/no/unit/nva/amazon/s3/AbortMultipartUploadHandler.java [hunk 3]
```diff
@@ -2,143 +2,103 @@
...
-import static no.unit.nva.amazon.s3.Environment.ALLOWED_ORIGIN_KEY;
-import static no.unit.nva.amazon.s3.Environment.MISSING_ENV_TEXT;
-import static no.unit.nva.amazon.s3.Environment.S3_UPLOAD_BUCKET_KEY;
-import static no.unit.nva.amazon.s3.GatewayResponse.ACCESS_CONTROL_ALLOW_ORIGIN;
-import static no.unit.nva.amazon.s3.GatewayResponse.BODY_KEY;
-import static org.apache.http.HttpHeaders.CONTENT_TYPE;
-import static org.apache.http.HttpStatus.SC_BAD_REQUEST;
-import static org.apache.http.HttpStatus.SC_INTERNAL_SERVER_ERROR;
-import static org.apache.http.HttpStatus.SC_NOT_FOUND;
+import no.unit.nva.amazon.s3.exception.InvalidInputException;
```
Comment (axthosarouris) [inline src/main/java/no/unit/nva/amazon/s3/AbortMultipartUploadHandler.java:8]: I am not sure if s3 is a good package name. Better to have a name that shows what is the role of underlying classes (publication storage?) and not the technology that we are using. If it generic storage maybe we should name it something else.

File: build.gradle (hunk 2 above)
Comment (janbj) [reply build.gradle:34]: I was using `MediaType.valueOf(input.getMimetype());
` and it gave me a class not found exception without that dependency.

File: src/main/java/no/unit/nva/fileupload/CreateUploadHandler.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+    @Override
+    protected CreateUploadResponseBody processInput(CreateUploadRequestBody input, RequestInfo requestInfo,
+                                                    Context context) throws ApiGatewayException {
+        validate(input);
+
+        String keyName = UUID.randomUUID().toString();
+        InitiateMultipartUploadRequest initRequest =
+                new InitiateMultipartUploadRequest(bucketName, keyName, toObjectMetadata(input));
+        InitiateMultipartUploadResult initResponse = s3Client.initiateMultipartUpload(initRequest);
```
Comment (janbj) [inline src/main/java/no/unit/nva/fileupload/CreateUploadHandler.java:91]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code manually manages uploads or downloads. The SDK can automatically handle these operations. Consider using the SDK to automatically handle this instead. SDK class: TransferManager. See [Using TransferManager for Amazon S3 Operations](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/examples-s3-transfermanager.html).

File: src/main/java/no/unit/nva/fileupload/ListPartsHandler.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        } catch (AmazonS3Exception e) {
+            throw new NotFoundException(S3_ERROR, e);
+        }
+
+        return listPartsElements;
+    }
+
+    private void validate(ListPartsRequestBody input) throws InvalidInputException {
+        try {
+            requireNonNull(input);
```
Comment (axthosarouris) [inline src/main/java/no/unit/nva/fileupload/ListPartsHandler.java:113]: same comment as previous Request

File: src/test/java/no/unit/nva/fileupload/AbortMultipartUploadHandlerTest.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        when(environment.readEnv(S3Constants.S3_UPLOAD_BUCKET_KEY)).thenReturn(S3Constants.S3_UPLOAD_BUCKET_KEY);
+        s3client = mock(AmazonS3Client.class);
+        abortMultipartUploadHandler = new AbortMultipartUploadHandler(environment, s3client, TEST_BUCKET_NAME);
+        context = new TestContext();
+        handlerUtils = new HandlerUtils(objectMapper);
+        outputStream = new ByteArrayOutputStream();
+    }
+
+    @Test
+    public void testHandleRequestMissingParameters() throws IOException {
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/fileupload/AbortMultipartUploadHandlerTest.java:62]: handleRequestReturnsBadRequestWhenPostBodyIsNull? Testing names should follow  the pattern
methodName(Returns/Throws)(Result/Exception)WhenInputIsSomething

File: src/test/java/no/unit/nva/fileupload/AbortMultipartUploadHandlerTest.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        abortMultipartUploadHandler.handleRequest(inputStream, outputStream, context);
+
+        GatewayResponse<SimpleMessageResponse> response = objectMapper.readValue(
+                outputStream.toByteArray(),
+                GatewayResponse.class);
+        assertEquals(SC_BAD_REQUEST, response.getStatusCode());
+    }
+
+    @Test
+    public void testHandleRequest() throws IOException {
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/fileupload/AbortMultipartUploadHandlerTest.java:74]: this is basically 2 tests. One for the code and one for the body. Renaming is required

File: src/test/java/no/unit/nva/fileupload/AbortMultipartUploadHandlerTest.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        assertEquals(SC_OK, response.getStatusCode());
+        assertNotNull(response.getBody());
+    }
+
+    private AbortMultipartUploadRequestBody abortMultipartUploadRequestBody() {
+        return new AbortMultipartUploadRequestBody(SAMPLE_UPLOAD_ID, SAMPLE_KEY);
+    }
+
+    @Test
+    public void testHandleFailingRequest() throws IOException {
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/fileupload/AbortMultipartUploadHandlerTest.java:92]: 2 tests. rename.

File: src/test/java/no/unit/nva/fileupload/CompleteUploadHandlerTest.java [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+        when(environment.readEnv(S3Constants.S3_UPLOAD_BUCKET_KEY)).thenReturn(S3Constants.S3_UPLOAD_BUCKET_KEY);
+        s3client = mock(AmazonS3Client.class);
+        completeUploadHandler = new CompleteUploadHandler(environment, s3client, TEST_BUCKET_NAME);
+        context = new TestContext();
+        handlerUtils = new HandlerUtils(objectMapper);
+        outputStream = new ByteArrayOutputStream();
+    }
+
+    @Test
+    public void testHandleRequestMissingParameters() throws IOException {
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/fileupload/CompleteUploadHandlerTest.java:73]: rename

File: src/test/java/no/unit/nva/fileupload/CompleteUploadHandlerTest.java [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+                outputStream.toByteArray(),
+                nva.commons.handlers.GatewayResponse.class);
+
+        assertEquals(SC_BAD_REQUEST, response.getStatusCode());
+    }
+
+
+
+    @Test
+    public void testHandleRequest() throws IOException {
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/fileupload/CompleteUploadHandlerTest.java:87]: refactor to 2 tests and rename.

File: src/test/java/no/unit/nva/fileupload/CompleteUploadHandlerTest.java [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+                outputStream.toByteArray(),
+                nva.commons.handlers.GatewayResponse.class);
+
+        assertNotNull(response);
+        assertEquals(SC_OK, response.getStatusCode());
+        assertNotNull(response.getBody());
+    }
+
+    @Test
+    public void testHandleFailingRequest() throws IOException {
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/fileupload/CompleteUploadHandlerTest.java:104]: refactor to two tests and rename

File: src/test/java/no/unit/nva/fileupload/CompleteUploadHandlerTest.java [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+                outputStream.toByteArray(),
+                nva.commons.handlers.GatewayResponse.class);
+
+        assertNotNull(response);
+        assertEquals(SC_NOT_FOUND, response.getStatusCode());
+        assertNotNull(response.getBody());
+    }
+
+    @Test
+    public void testHandleRequestWithEmptyElement() throws IOException {
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/fileupload/CompleteUploadHandlerTest.java:121]: rename

File: src/main/java/no/unit/nva/fileupload/CreateUploadHandler.java [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+
+        return new CreateUploadResponseBody(initResponse.getUploadId(), keyName);
+    }
+
+    private void validate(CreateUploadRequestBody input) throws InvalidInputException {
+        try {
+            requireNonNull(input);
+            requireNonNull(input.getFilename());
```
[the remark this answers is not in the document]
Comment (janbj) [reply src/main/java/no/unit/nva/fileupload/CreateUploadHandler.java:99]: I can do this in next part of refactoring when I'm splitting into subprojects.

**Pull request merged**